### ISSUE TITLE
Rule block-scoped-var fires if a variable is reused

### ISSUE THREAD

**Issue by pauljansen42** [eslint/eslint#10613]
**Tell us about your environment**

* **ESLint Version:** 5.1.0
* **Node Version:**
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:** I have only configured rule block-scoped-var

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
  for (var i = 0; i < 3; i++) {
  }
  for (var i = 0; i < 3; i++) {
  }
```

```bash
I ran it with the online tool
```

**What did you expect to happen?**
no violations

**What actually happened? Please include the actual, raw output from ESLint.**
violations of rule block-scoped-var. This is because variable "i" is declared twice.
